core/trino-main/src/main/java/io/trino/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -338,6 +338,10 @@ public StreamProperties visitExchange(ExchangeNode node, List<StreamProperties>
                     if (node.getPartitioningScheme().getPartitioning().getHandle().equals(FIXED_ARBITRARY_DISTRIBUTION)) {
                         return new StreamProperties(FIXED, Optional.empty(), false);
                     }
+                    // empty arguments list means the bucketing function is effectively constant (1 bucket)
+                    if (node.getPartitioningScheme().getPartitioning().getArguments().isEmpty()) {
+                        return new StreamProperties(SINGLE, Optional.of(ImmutableSet.of()), false);
```

Comment:
there is a shortcut for it: `StreamPropertyDerivations.StreamProperties#singleStream`